### PR TITLE
fix: [CDS-44070]: indicate through label that region field is optional in the last step of Amazon S3 artifact

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- indicate through label that region field is optional in the last step of Amazon S3 artifact
- Add test cases for region field that were not written as region field was added after initial dev effort of Amazon S3 artifact

#### Screenshots
Screenshot 2022-10-03 at 12 38 58 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11613)

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonS3Artifact/__tests__/__snapshots__/AmazonS3.test.tsx.snap [hunk 1]
```diff
@@ -36,7 +36,7 @@ exports[`AmazonS3 tests bucket name field is Runtime input when connectorId from
                     class="Tooltip--acenter"
                     data-tooltip-id="artifactoryArtifact_region"
                   >
-                    regionLabel
+                    optionalField
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonS3Artifact/__tests__/__snapshots__/AmazonS3.test.tsx.snap:39]: In snapshot, only optionalField is there?
Should it be `optionalField regionLabel`

Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonS3Artifact/__tests__/__snapshots__/AmazonS3.test.tsx.snap:39]: It does not work that way. Only displays main text. name is passed as var and that is not displayed anywhere.

**Comment by chetanvekariya-harness** [comment]
enhancement/amazon_S3

**Comment by chetanvekariya-harness** [comment]
retrigger titlecheck

**Review by Tamaraius — APPROVED** [review]

**Review by depanshu-dhiman — APPROVED** [review]

**Pull request merged**